### DISCUSSION TITLE
openslides not reachable

### DISCUSSION THREAD

**Discussion by mvote** [OpenSlides/OpenSlides#6485] [Q&A]
Hi! After "playing" with openslides for a while, I would like to use version 3.4 for a real assembly of a small non profit organization. 
But unfortunately I am not able to configure the production version in a correct way. It is is not reachable from the outside.
  
I installed the production dockerized 3.4 version according the manual on a virtual Debian 11 machine, without mayor problems. Certificates are also available. "Docker compose up" starts without any issues and the docker-proxy outputs "Configured https" put it is not reachable from the outside. The real reachable IP (eth0) is 10.1.0.4 and this is also set as instance domain at the .env file (and port 443 as EXTERNAL_HTTPS_PORT). These values do also appear in the docker-compose.yml file after m4. 
Anyhow when I execute a nmap port scan on 10.1.0.4 on the host machine no service listens at 443 (which explains why openslides is not reachable). But there is a service active on port 443 at 127.0.0.1, when the container is running. I tried to change the instance domain to 0.0.0.0 without success. Maybe I have to bridge port 443 from the outside to the container? But I guess this should be done by the proxy. Well, I've to admit that I am not a docker expert. 

Maybe someone of you has an idea? Thank you very much for the great work and time you are investing in openslides. 

Kind regards
Matthias

**Comment by mvote** [comment]
Hi! 
I can answer it now by myself. 
Just needed to change docker-compose.yml. Under Services: proxy: ports: the proxy server is configured. Here I found the 127.0.0.1 setting giving the proxy the command to bind to 127.0.0.1. Not so clever, if not an exclusive local installation is needed. Well I guess it would be nice to document it somewhere, in the installation chapter ... but I also understand, why not. 
Maybe this helps someone else. 

CU Matthias

**Reply by ostcar** [reply]
It seems, it is the same for OS4.0: https://github.com/OpenSlides/openslides-manage-service/blob/77ed8334e46b5ea8112735c3d4f5ee1793e7d982/pkg/config/default-config.yml#L6

The default is to listen to 127.0.0.1. But it might be easier to set another host in the config.yml.

Maybe the idea is not to open a public port as default. If you have a webserver running on your host system, then you will probably want to proxy from a specific domain to the local OpenSlides-Port.

But as far as I can see, there is currently no documentation anywhere how to make OpenSlides public (with a proxy like nginx or apache or by setting Host to an empty string). @normanjaeckel Maybe you want to write something in the README.md?

**Reply by mvote** [reply]
Hi Oscar! When you use the docker version then there is a web server installed (caddy). And this server handles also the SSL certificates and so on (and runs in a separate container). But in the standard config the Webserver is configured in a way that it does only the 443 to 8000 conversion and not from public IP to private IP (127.0.0.1), at least not in my case. That is what I did (in my second answer). By the way now I changed the m4 file and not the yml anymore. Which is more elegant, because now I do not loose the configuration every time when I change the .env.
